DylibBinder/CheckInventory.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using SwiftReflector.Inventory;
+
+namespace DylibBinder {
+	public class CheckInventory {
+		public CheckInventory (ModuleInventory mi)
+		{
+			Classes = new List<ClassContents> ();
+			Structs = new List<ClassContents> ();
+			Enums = new List<ClassContents> ();
+			Protocols = new List<ProtocolContents> ();
+
+			GetClassesStructsEnums (mi);
+			GetProtocols (mi);
+		}
+
+		public List<ClassContents> Classes { get; }
+		public List<ClassContents> Structs { get; }
+		public List<ClassContents> Enums { get; }
+		public List<ProtocolContents> Protocols { get; }
+
+		void GetClassesStructsEnums (ModuleInventory mi)
+		{
+			foreach (var m in mi.ModuleNames) {
+				foreach (var elem in mi.ClassesForName (m)) {
+					if (!elem.Name.ToString ().IsPublic ())
+						continue;
+					if (elem.Name.IsClass)
+						Classes.Add (elem);
+					else if (elem.Name.IsStruct)
+						Structs.Add (elem);
+					else if (elem.Name.IsEnum)
+						Enums.Add (elem);
+				}
+			}
+			SortNominalTypeLists (null, Classes, Structs, Enums);
+		}
+
+		void GetProtocols (ModuleInventory mi)
+		{
+			foreach (var m in mi.ModuleNames) {
+				foreach (var p in mi.ProtocolsForName (m)) {
+					if (!p.Name.ToString ().Contains ("_"))
+						Protocols.Add (p);
+				}
+			}
+			SortNominalTypeLists (Protocols);
+		}
+
+		void SortNominalTypeLists (List<ProtocolContents> protocolList, params List<ClassContents> [] nominalTypeLists)
```

Comment:
Uh, I'm going to make you think about this one. You can write cleaner. We have two possible options that I need you to investigate:

1. Check in ProtocolContents and ClassContents have a common interface (they both have the property Name. If that is the case, you can rewrite the methods as the following and just sort no matter the T:

```csharp
void SortNominalTypeLists<T> (params List<T> [] nominalTypeLists) where T : IMyInterface
```

2. If we do not have a common interface, do method overloading. Calling it with the first parameter as null is really confusing. I saw it and I had to check the method signature, the 'contract' of our API is ugly by forcing you to know what is the type of the first param (and it being diff than the rest).
```csharp
void SortNominalTypeLists (params List<ProtocolContents> protocolList)
void SortNominalTypeLists (params List<ClassContents> [] nominalTypeLists)
```